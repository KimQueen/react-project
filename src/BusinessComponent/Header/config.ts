export const tabConfig = [
  {
    name: '首页',
    url: '/',
  },
  {
    name: '关于',
    url: '/',
  },
  {
    name: '联系我们',
    url: '/',
  },
];
