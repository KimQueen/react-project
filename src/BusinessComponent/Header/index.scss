.warpContent-header {
  position: sticky;
  top: 0;
  height: 64px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  box-shadow:
    0 1px 2px 0 rgba(0, 0, 0, 0.03),
    0 1px 6px -1px rgba(0, 0, 0, 0.02),
    0 2px 4px 0 rgba(0, 0, 0, 0.02);

  .left {
    width: 256px;
    border-right: 1px solid #e8e8e8;

    .imgStyle {
      position: absolute;
      left: -2px;
      top: -13px;
      width: 98px;
    }

    .title {
      font-size: 25px;
      margin-left: 101px;
      padding-top: 19px;
    }
  }

  .right {
    padding: 24px;
    display: flex;
    // width: 100px;

    .tabItem {
      cursor: pointer;
      width: 80px;
    }

    .tabItem:hover {
      width: 80px;
      color: #0958d9;
    }

    .gitImg {
      width: 32px;
      height: 32px;
      margin-top: -4px;
    }
  }
}
