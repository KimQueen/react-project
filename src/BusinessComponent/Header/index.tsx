import React from 'react';
import { useNavigate } from 'react-router-dom';
import imgURL from '../../assets/header.png';
import gitUrl from '../../assets/git.jpeg';
import { tabConfig } from './config';
import './index.scss';

const Header = () => {
  const navigate = useNavigate();

  const handleClick = (url: string) => {
    navigate(url);
  };

  const gotoGitPage = () => {
    window.open('https://github.com/KimQueen/react-project');
  };
  return (
    <div className="warpContent-header">
      <div className="left">
        <img src={imgURL} alt="图标" className="imgStyle" />
        <div className="title">Component</div>
      </div>
      <div className="content" />
      <div className="right">
        {tabConfig.map((item) => (
          <div onClick={() => handleClick(item.url)} className="tabItem">
            {item.name}
          </div>
        ))}
        <div onClick={() => gotoGitPage()}>
          <img src={gitUrl} className="gitImg" alt="git" />
        </div>
      </div>
    </div>
  );
};

export default Header;
