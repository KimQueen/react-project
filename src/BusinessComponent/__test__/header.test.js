import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Header from '../Header';

const mockUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUsedNavigate,
}));

describe('Header', () => {
  test('renders Header component', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
