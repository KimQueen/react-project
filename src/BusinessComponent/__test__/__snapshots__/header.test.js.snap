// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header component 1`] = `
<DocumentFragment>
  <div
    class="warpContent-header"
  >
    <div
      class="left"
    >
      <img
        alt="图标"
        class="imgStyle"
        src="header.png"
      />
      <div
        class="title"
      >
        Component
      </div>
    </div>
    <div
      class="content"
    />
    <div
      class="right"
    >
      <div
        class="tabItem"
      >
        首页
      </div>
      <div
        class="tabItem"
      >
        关于
      </div>
      <div
        class="tabItem"
      >
        联系我们
      </div>
      <div>
        <img
          alt="git"
          class="gitImg"
          src="git.jpeg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
